packages/autocomplete-plugin-recent-searches/src/createPlugin.ts

```diff
@@ -0,0 +1,83 @@
+import { AutocompletePlugin } from '@algolia/autocomplete-core';
+
+import { createRecentSearchStore } from './createRecentSearchStore';
+
+type PluginOptions = {
+  /**
+   * The number of searches to store.
+   *
+   * @default 5
+   */
+  limit?: number;
+
+  /**
+   * The key to distinguish multiple stores of recent searches.
+   *
+   * @default ''
+   * @example
+   * // 'top_searchbar'
+   */
+  key?: string;
+};
+
+type RecentSearchesPlugin<TItem> = AutocompletePlugin<TItem> & {
+  getFacetFilters: () => string[];
+};
+
+export function createPlugin<TItem>({
+  limit = 5,
+  key = '',
+}: PluginOptions = {}): RecentSearchesPlugin<TItem> {
+  const store = createRecentSearchStore({
+    key: ['RECENT_SEARCHES', key].filter(Boolean).join(_),
```

Comment:
When refactored, we should have a separator that is different from our key to allow easier parsing.

```suggestion
    key: ['RECENT_SEARCHES', key].filter(Boolean).join('__'),
```

Reply:
done https://github.com/algolia/autocomplete.js/pull/316/commits/90fc79061c0beeb95cceedd446d8f1e2445a8b21